calculator_test.go

```diff
@@ -2,23 +2,49 @@ package calculator_test
 
 import (
 	"calculator"
+	"fmt"
 	"testing"
 )
 
+type testcase struct {
+	a    float64
+	b    float64
+	want float64
+}
+
 func TestAdd(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
+
+	tests := []testcase{
+		{a: 2, b: 2, want: 4},
+		{a: -1, b: 1, want: 0},
+	}
+
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("test case %d", i), func(t *testing.T) {
```

Comment:
It'll be handy to give each case a `name` field that we can use here—a good way to name test cases is to say what's unique about this case. For example, `"Sum of two positives is positive"`, or `"A negative and a positive that sum to zero"`.

Reply:
+1. I'd stopped work at this point but this was my next step.

Reply:
A follow on question - in the past I've declared the test case struct as a `map[string]struct{.....}` and then the key of the map is the test name. Do you have a preference for either style, and if so why ?

Reply:
That's an interesting question! I've never thought about doing that before. I can't honestly see any good reason to do it; our test needs only to iterate over the cases, not to look up cases by name. So while there's no strong argument against it, I can't see an argument _for_ it either!